handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -63,6 +65,32 @@
      */
     static final int NOT_ENCRYPTED = -2;
 
+    static final int MAX_PLAINTEXT_LENGTH = 16 * 1024; // 2^14
+
+    /**
+     * This is the maximum overhead when encrypting plaintext as defined by
+     * <a href="https://www.ietf.org/rfc/rfc5246.txt">rfc5264</a>,
+     * <a href="https://www.ietf.org/rfc/rfc5289.txt">rfc5289</a> and openssl implementation itself.
+     *
+     * Please note that we use a padding of 16 here as openssl uses PKC#5 which uses 16 bytes while the spec itself
+     * allow up to 255 bytes. 16 bytes is the max for PKC#5 (which handles it the same way as PKC#7) as we use a block
+     * size of 16. See <a href="https://tools.ietf.org/html/rfc5652#section-6.3">rfc5652#section-6.3</a>.
+     *
+     * 16 (IV) + 48 (MAC) + 1 (Padding_length field) + 15 (Padding) + 1 (ContentType) + 2 (ProtocolVersion) + 2 (Length)
+     *
+     * TODO: We may need to review this calculation once TLS 1.3 becomes available.
+     */
+    static final int MAX_ENCRYPTION_OVERHEAD_LENGTH = 15 + 48 + 1 + 16 + 1 + 2 + 2;
```

Comment:
I actually removed this stuff [1] and put it into tcnative because the OpenSSL implementation may be able to provide us a more exact answer [2]. Can we push this into conscrypt too?

[1] https://github.com/netty/netty/pull/6365/files#diff-cfd417915faad637e4278b54adafd9baL114
[2] https://github.com/netty/netty-tcnative/pull/233/files#diff-54594b1ed814f0d6cfbe0cd7893e43d6R1659